plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3129,6 +3129,45 @@ public void testOptimizeForPartitionedTable()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @DataProvider
+    public Object[][] tableFormatVersion()
+    {
+        return new Object[][] {
+                {1},
+                {2},
+        };
+    }
+
+    @Test
+    public void testOptimizeTableFormatVersion2()
```

Comment:
Could you please write similar test or modify this in such a way that deletion is not the last operation ? So delete files are not part of the `current snapshot`?